### PR TITLE
feat(): initial version of ts basics.

### PR DESCRIPTION
**Pull request by MahtaMirzaei**
Related: #6

### CODE REVIEW

File: Mahta/TS/TS-basics/basics.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
+function add(a: number, b: number): number {
+  return a + b;
+}
+
+function flipACoin(): string {
+  const random: number = Math.random();
+  if (random > 0.5) {
+      return "tail";
```
Comment (fatemeh-ra) [inline Mahta/TS/TS-basics/basics.ts:8]: define an enum for return type of the function

File: Mahta/TS/TS-basics/basics.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  } else {
+      return "head";
+  }
+}
+
+function concat(stringA: string, stringB: string): string {
+  return stringA.concat(stringB);
+}
+
+function power2(x: number): number {
```
Comment (fatemeh-ra) [inline Mahta/TS/TS-basics/basics.ts:18]: return type of power2 should be bigint

File: Mahta/TS/TS-basics/basics.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  return 2 ** x;
+}
+
+function getUser(firstName: string, lastName: string): { firstName: string; lastName: string } {
+  const capitalizedFirstName = firstName.replace(firstName.charAt(0), firstName.charAt(0).toUpperCase());
```
Comment (fatemeh-ra) [inline Mahta/TS/TS-basics/basics.ts:23]: also use lower case function for the rest of characters

File: Mahta/TS/TS-basics/basics.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  const intervalId = setInterval(() => {
+      console.log('Time: ' + new Date());
+      counter++;
+      if (counter === Math.floor(n / x)) {
+          clearInterval(intervalId);
+      }
+  }, x * 1000);
+}
+
+function callFuncNTimes(func: Function, n: number, ...funcParams: any[]): void {
```
Comment (fatemeh-ra) [inline Mahta/TS/TS-basics/basics.ts:51]: our function doesn't have any input parameters.  
try this with flipACoin as input.

File: Mahta/TS/TS-basics/basics.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  for (let i = 0; i < n; i++) {
+      func(...funcParams);
+  }
+}
+
+function isNil(param: any): boolean {
+  return !param;
+}
+
+function callPropInObj(object: any, propertyName: string): void {
```
Comment (fatemeh-ra) [inline Mahta/TS/TS-basics/basics.ts:61]: don't use any here. It is meant to be a generic type function.

File: Mahta/TS/TS-basics/basics.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  object?.[propertyName]?.();
+}
+
+function isGmail(email: string): boolean {
+  return email.endsWith("@isGmail.com");
+}
+
+function duplicateArray(): string[] {
+  let fruits: string[] = ['Apple', 'Orange', 'Strawberry'];
```
Comment (fatemeh-ra) [inline Mahta/TS/TS-basics/basics.ts:70]: the array should be the input of the function.

File: package.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (fatemeh-ra) [inline package.json:1]: why do you have another package.json & package-lock.json out of your directory??

File: Mahta/TS/TS-basics/.gitignore [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+# Windows
+Thumbs.db
+
+# IDE/Editor specific files
+.vscode/
+.idea/
+
+# Package lock and yarn lock files
+package-lock.json
```
Comment (fatemeh-ra) [inline Mahta/TS/TS-basics/.gitignore:31]: package-lock.json should not be ignored. it's important to use the exact same package versions at different run times.

File: node_modules/.package-lock.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fatemeh-ra) [inline node_modules/.package-lock.json:1]: what is it exactly? how did you ignore node_mosules and include this file in your commit?

**Review by fatemeh-ra — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (MahtaMirzaei) [reply package.json:1]: one belongs to node modules. how can I fix that> should I mix them?

File: node_modules/.package-lock.json (hunk 9 above)
Comment (MahtaMirzaei) [reply node_modules/.package-lock.json:1]: fixed that.

File: package.json (hunk 7 above)
Comment (MahtaMirzaei) [reply package.json]: found out what was wrong, removed all

File: Mahta/TS/TS-basics/basics.ts [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+  return !param;
+}
+
+function callPropInObj<T>(object: T, propertyName: keyof T): void {
+  (object[propertyName] as unknown as () => void)?.();
+}
+
+
+function isGmail(email: string): boolean {
+  return email.endsWith("@isGmail.com");
```
Comment (fatemeh-ra) [inline Mahta/TS/TS-basics/basics.ts:78]: what?? isGmail.com?

**Review by fatemeh-ra — CHANGES_REQUESTED** [review]

**Pull request merged**